app/src/androidTest/java/com/github/se/endToEnd/ReceiveRequestFlowEndToEnd.kt

```diff
@@ -0,0 +1,268 @@
+import com.github.se.icebreakrr.Icebreakrr
+import com.github.se.icebreakrr.model.message.MeetingRequestService
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.messaging.RemoteMessage
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@HiltAndroidTest
+class ReceiveRequestFlowEndToEnd {
+  val intent =
+      Intent(InstrumentationRegistry.getInstrumentation().targetContext, Icebreakrr::class.java)
+
+  @Inject lateinit var authInjected: FirebaseAuth
+
+  @Inject lateinit var firestoreInjected: FirebaseFirestore
+
+  @Inject lateinit var authStateListener: FirebaseAuth.AuthStateListener
+
+  private lateinit var message: RemoteMessage.Builder
+  private lateinit var service: MeetingRequestService
+
+  @get:Rule var hiltRule = HiltAndroidRule(this)
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setup() {
+    hiltRule.inject()
+    intent.putExtra("IS_TESTING", true)
+    ActivityCompat.setPermissionCompatDelegate(TestPermissionDelegate())
+
+    service = MeetingRequestService()
+    message =
+        RemoteMessage.Builder("test_sender")
+            .setMessageId("12345")
+            .addData("senderUID", "0")
+            .addData("senderName", "Alice Inwonderland")
+  }
+
+  @Test
+  fun receiveRequestDeclineFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      composeTestRule.waitForIdle()
+
+      // decline and check that everything updates
+      composeTestRule.onNodeWithTag("declineButton").performClick()
+      composeTestRule.waitForIdle()
+      composeTestRule
+          .onNodeWithTag("badgeNotification", useUnmergedTree = true)
+          .assertIsNotDisplayed()
+      composeTestRule.onNodeWithTag("badgeHeatmap").assertIsNotDisplayed()
+
+      // go on heatmap
+      composeTestRule.onNodeWithTag("navItem_Map").performClick()
+      scenario.close()
+    }
+  }
+
+  @Test
+  fun receiveRequestTimeCancellationFlowEndToEnd() {
+    ActivityScenario.launch<Icebreakrr>(intent).use { scenario ->
+      receiveRequestAndCheckProfile()
+
+      // receives a cancellation for time :
+      message.addData("message", "TIME").addData("title", "MEETING CANCELLATION")
+      service.onMessageReceived(message.build())
+
+      composeTestRule.waitForIdle()
+
+      // check that alice disappear :
+      composeTestRule.onNodeWithTag("profileCard").assertIsNotDisplayed()
+      composeTestRule
+          .onNodeWithTag("badgeNotification", useUnmergedTree = true)
+          .assertIsNotDisplayed()
+      scenario.close()
+    }
+  }
+
+  @Test
+  fun receiveRequestDistanceCancellationFlowEndToEnd() {
```

Comment:
Nitpick: The name is lengthy and not entirely clear. Change it or add kdoc for better code clarity.

Reply:
I added a _and_ since it is literally receiving a request, then receiving a distance cancellation and check that everything is coherent